### PR TITLE
DOCS-1429: Document Robot API remaining methods

### PR DESCRIPTION
**Pull request by sguequierre**
* `WithChannel()` in particular needs SME review.
* Add undocumented methods.
* Removed in checker `get_component` and `get_services` methods instead of documenting, bc they are an alternative method to `from_robot` that is meant to barely be called if ever.

### CODE REVIEW

File: docs/build/program/apis/robot.md [hunk 1]
```diff
@@ -170,6 +170,313 @@ For the full list of methods, see the [Viam Python SDK documentation](https://py
...
+**Returns:**
+
+- `RobotClient`: The robot client.
+
+```python {class="line-numbers linkable-line-numbers"}
+from grpclib.testing import ChannelFor
+from viam.robot.service import RobotService
+
+async with ChannelFor([RobotService]) as channel:
+      robot = await RobotClient.with_channel(channel, RobotClient.Options())
```
Comment (sguequierre) [inline docs/build/program/apis/robot.md:264]: @cheukt Is this accurate? How can I show the channels between two robots with Python? grabbed this from test code but doesn't look like it's making an actual channel

File: docs/build/program/apis/robot.md [hunk 2]
```diff
@@ -170,6 +170,313 @@ For the full list of methods, see the [Viam Python SDK documentation](https://py
...
+**Returns:**
+
+- `RobotClient`: The robot client.
+
+```python {class="line-numbers linkable-line-numbers"}
+from grpclib.testing import ChannelFor
+from viam.robot.service import RobotService
+
+async with ChannelFor([RobotService]) as channel:
+    robot = await RobotClient.with_channel(channel, RobotClient.Options())
```
Comment (sguequierre) [inline docs/build/program/apis/robot.md:264]: @cheukt Is this accurate? How can I show the channels between two robots with Python? grabbed this from test code but doesn't look like it's making an actual channel

File: docs/build/program/apis/robot.md (hunk 1 above)
Comment (cheukt) [reply docs/build/program/apis/robot.md]:
```
from viam.robot.client import RobotClient
from viam.rpc.dial import DialOptions, dial

async def connect_with_channel() -> RobotClient:
    async with await dial('ADDRESS', DialOptions() as channel:
        return await RobotClient.with_channel(channel, RobotClient.Options())

robot = await connect_with_channel()
print(robot.resource_names)
await robot.close()
```
would be how you get a channel in the Python SDK

File: docs/build/program/apis/robot.md [hunk 3]
```diff
@@ -170,6 +170,313 @@ For the full list of methods, see the [Viam Python SDK documentation](https://py
...
+from viam.robot.service import RobotService
+
+async with ChannelFor([RobotService]) as channel:
+      robot = await RobotClient.with_channel(channel, RobotClient.Options())
+```
+
+For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/viam/robot/client/index.html#viam.robot.client.RobotClient.with_channel).
+
+{{% /tab %}}
+{{< /tabs >}}
```
Comment (cheukt) [inline docs/build/program/apis/robot.md]: documenting these methods here feels weird - these are specific Python library methods that are not part of the API and so would be different across languages. maybe it should be in a different section?

**Review by npentrel — APPROVED** [review]
Lgtm once Cheuk approves

File: docs/build/program/apis/robot.md (hunk 1 above)
Comment (sguequierre) [reply docs/build/program/apis/robot.md]: @cheukt got it thank you!

File: docs/build/program/apis/robot.md [hunk 4]
```diff
@@ -170,6 +170,316 @@ For the full list of methods, see the [Viam Python SDK documentation](https://py
...
+
+**Returns:**
+
+- `RobotClient`: The robot client.
+
+```python {class="line-numbers linkable-line-numbers"}
+from viam.robot.client import RobotClient
+from viam.rpc.dial import DialOptions, dial
+
+async def connect_with_channel() -> RobotClient:
```
Comment (sguequierre) [inline docs/build/program/apis/robot.md:263]:
```suggestion

async def connect_with_channel() -> RobotClient:
```

File: docs/build/program/apis/robot.md [hunk 5]
```diff
@@ -170,6 +170,321 @@ For the full list of methods, see the [Viam Python SDK documentation](https://py
...
+**Returns:**
+
+- [(RobotClient.Options)](https://python.viam.dev/autoapi/viam/robot/client/index.html#viam.robot.client.RobotClient.Options): Options for connecting the `RobotClient`.
+
+```python
+# Replace "<API-KEY>" (including brackets) with your robot's api key
+api_key='<API-KEY>'
+# Replace "<API-KEY-ID>" (including brackets) with your robot's api key
+# id
+api_key_id='<API-KEY-ID>'
```
Comment (sguequierre) [inline docs/build/program/apis/robot.md:196]:
```suggestion
api_key_id  = '<API-KEY-ID>'
```

File: docs/build/program/apis/robot.md [hunk 6]
```diff
@@ -170,6 +170,321 @@ For the full list of methods, see the [Viam Python SDK documentation](https://py
...
+- `api_key_id` [(str)](https://docs.python.org/3/library/stdtypes.html#text-sequence-type-str): Your API key ID.
+  Must be a valid UUID.
+
+**Returns:**
+
+- [(RobotClient.Options)](https://python.viam.dev/autoapi/viam/robot/client/index.html#viam.robot.client.RobotClient.Options): Options for connecting the `RobotClient`.
+
+```python
+# Replace "<API-KEY>" (including brackets) with your robot's api key
+api_key='<API-KEY>'
```
Comment (sguequierre) [inline docs/build/program/apis/robot.md:193]:
```suggestion
api_key = '<API-KEY>'
```

**Pull request merged**